### DISCUSSION TITLE
3D basemap in pydeck

### DISCUSSION THREAD

**Discussion by raghulkrishna** [visgl/deck.gl#5716] [Q&A]
how do i implement this in pydeck?

```
 map.on('load', () => {
    map.addLayer({
      id: '3d-buildings',
      source: 'composite',
      'source-layer': 'building',
      filter: ['==', 'extrude', 'true'],
      type: 'fill-extrusion',
      minzoom: 14,
      paint: {
        'fill-extrusion-color': '#ccc',
        'fill-extrusion-height': ['get', 'height']
      }
    });

    renderLayers(map, data);
  });
```

**Comment by ajduberstein** [comment]
Is there a particular deck.gl example you're referencing? If yes, please link it here.

If you're interested in extrusion, you can simply set `extruded=True` on any layer that supports it (e.g. a `HexagonLayer`, `ColumnLayer`, `GeoJsonLayer`, e.g. [here](https://deckgl.readthedocs.io/en/latest/gallery/geojson_layer.html)). If you're interested in a base map that has 3D buildings, [you can build a style URL in Mapbox](https://docs.mapbox.com/help/tutorials/add-3d-buildings-studio/) and pass a Mapbox URL to pydeck (see [`map_style`](https://pydeck.gl/deck.html) in the docs).
